src/main/java/seedu/address/model/person/Gender.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Represents a Person's gender in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_CONSTRAINTS = "Gender is an optional field, it should be one of the following"
+        + " format: m, f, M, F, male, female, Male, Female";
+
+    private static final List<String> VALID_GENDERS = new ArrayList<>(Arrays.asList("m", "male", "M", "Male",
```

Comment:
Not sure if `ArrayList` is the best data structure to use here. Since you're only using it as lookup table to check if a gender is valid, perhaps a `HashSet` would be better suited instead? A `HashSet` allows you to perform direct lookups in O(1), while an `ArrayList` has to search the entire list to find the item. Although, the performance advantage doesn't really matter here since the `ArrayList` is a fixed size, so it's also O(1) lookup.

Alternatively, you could also use regex strings to match against the user input, though the regex expression may be a little complicated since there are many valid genders.